core/src/main/scala/org/http4s/blaze/channel/nio2/ClientChannelFactory.scala

```diff
@@ -21,7 +23,16 @@ import scala.util.control.NonFatal
 final class ClientChannelFactory(
     bufferSize: Int = DefaultBufferSize,
     group: Option[AsynchronousChannelGroup] = None,
-    channelOptions: ChannelOptions = ChannelOptions.DefaultOptions) {
+    channelOptions: ChannelOptions = ChannelOptions.DefaultOptions,
+    scheduler: TickWheelExecutor = Execution.scheduler,
+    connectingTimeout: Option[FiniteDuration] = None) {
+
+  // for binary compatibility with <=0.14.6
```

Comment:
Nicely done.  (This incidentally is why http4s uses the builder pattern a lot in its backends.  It makes adding new options easier.)